tests/tests/tier0/bluechi-is-online-system-wait/test_bluechi_is_online_system_wait.py

```diff
@@ -0,0 +1,210 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+import threading
+import time
+from typing import Dict
+
+from bluechi_test.bluechi_is_online import BluechiIsOnline
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+from bluechi_test.util import Timeout, get_test_env_value_int
+
+LOGGER = logging.getLogger(__name__)
+
+AGENT_ONE = "agent-one"
+AGENT_TWO = "agent-two"
+IMMEDIATE_RETURN_TIMEOUT_MS = get_test_env_value_int("IMMEDIATE_RETURN_TIMEOUT", 1000)
+WAIT_PARAM_VALUE_MS = get_test_env_value_int("WAIT_PARAM_VALUE", 5000)
+SLEEP_DURATION = get_test_env_value_int("SLEEP_DURATION", 2)
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_system(
+    bluechi_is_online: BluechiIsOnline,
+    wait_time: int,
+    future: ResultFuture,
+):
+    future.result = bluechi_is_online.system_is_online(
+        wait_time,
+    )
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    # Test 1: Both agents and controller are online
+    LOGGER.debug("Testing with both agents and controller online.")
+    system_result_test_one = ResultFuture()
+    system_thread_wait = threading.Thread(
+        target=check_system,
+        args=(
+            ctrl.bluechi_is_online,
+            IMMEDIATE_RETURN_TIMEOUT_MS,
+            system_result_test_one,
+        ),
+    )
+    system_thread_wait.start()
+    system_thread_wait.join()
+    assert (
+        system_result_test_one.result
+    ), "System wait command should not produce output when system is running."
+
+    # Test 2: Agent-one offline
+    LOGGER.debug("Starting test number 2 - agent one should remain offline.")
+    LOGGER.debug("Stopping agent-one.")
+    nodes[AGENT_ONE].systemctl.stop_unit("bluechi-agent")
+    assert nodes[AGENT_ONE].wait_for_unit_state_to_be("bluechi-agent", "inactive")
+
+    start_time = time.time()
```

Comment:
Shouldn't it look similar to https://github.com/eclipse-bluechi/bluechi/blob/main/tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py#L53 ?

```python
    start_time = time.time()
    result = ctrl.bluechi_is_online.system_is_online(wait=WAIT_PARAM_VALUE)
    assert (
        not result
    ), f"Expected bluechi-is-online with --wait={WAIT_PARAM_VALUE} to return an error"
    assert (
        time.time() - start_time > WAIT_PARAM_VALUE / 1000
    ), f"Expected around {WAIT_PARAM_VALUE} seconds for bluechi-is-online to exit"
```